source/src/scene/GameplayScene.java

```diff
@@ -365,6 +363,7 @@ public void drawUI() {
 			case MODIFY :
 				editorItem.update();
 				editorItem.display();
+				magic.relocate();
```

Comment:
I would move this function call into the magic item's update / display function.

In the magic source class the function will look something like:

@Override
public void display() {
	trail.run();
	this.relocate();
}